lib/docker/executeDockerBuild.ts

```diff
@@ -256,47 +258,61 @@ async function dockerPush(images: string[],
                           options: DockerOptions,
                           gi: ProjectAwareGoalInvocation): Promise<ExecuteGoalResult> {
 
-    let result = Success;
-
     if (await pushEnabled(gi, options)) {
 
-        // 1. run docker push
-        for (const image of images) {
-            result = await gi.spawn(
-                "docker",
-                ["push", image],
-                {
-                    env: {
-                        ...process.env,
-                        DOCKER_CONFIG: dockerConfigPath(options, gi.goalEvent),
+        // Login to registry(s)
+        await Promise.all(
+            options.registries.map(async r => {
+                const loginResult = await dockerLogin(r, options, gi);
+                if (loginResult.code !== 0) {
+                    throw new Error(`Failed login for registry => ${r.url}!`);
+                }
+            }),
+        );
+
+        await Promise.all(
```

Comment:
Please change to sequential as otherwise you'll potentially create a large number of concurrent pushes and also the log becomes unreadable.

Reply:
As discussed, this is intentional.  Will add a config option to control if these are parallel or not.